common/analysis/lint_rule_status.cc

```diff
@@ -58,6 +58,42 @@ void LintStatusFormatter::FormatLintRuleStatus(std::ostream* stream,
   }
 }
 
+struct LintViolationWithStatus {
+  const LintViolation *violation;
+  const LintRuleStatus *status;
+
+  LintViolationWithStatus(const LintViolation *v, const LintRuleStatus *s)
+      : violation(v), status(s) {}
+
+  bool operator<(const LintViolationWithStatus& r) const {
+    return (violation->token.text.data() < r.violation->token.text.data());
+  };
+};
+
+void LintStatusFormatter::FormatLintRuleStatuses(
+    std::ostream* stream,
+    const std::vector<LintRuleStatus> &statuses,
+    absl::string_view base,
+    absl::string_view path) const {
+
+  std::set<LintViolationWithStatus> violations;
+
+  for (auto &status : statuses) {
```

Comment:
// TODO(fangism): rewrite as a linear time merge of pre-ordered sub-sequences

Reply:
Sure, added